index.js

```diff
@@ -78,67 +81,34 @@ function lintFiles (files, opts, cb) {
   if (typeof files === 'string') files = [ files ]
   if (files.length === 0) files = DEFAULT_PATTERNS
 
-  // traverse filesystem
-  parallel(files.map(function (pattern) {
-    return function (cb) {
-      glob(pattern, {
-        cwd: opts.cwd,
-        ignore: opts._ignore,
-        nodir: true
-      }, cb)
-    }
-  }), function (err, results) {
+  deglob(files, opts, function (err, allFiles) {
     if (err) return cb(err)
-
-    // flatten nested arrays
-    var files = results.reduce(function (files, result) {
-      result.forEach(function (file) {
-        files.push(path.resolve(opts.cwd, file))
-      })
-      return files
-    }, [])
-
-    // de-dupe
-    files = uniq(files)
-
-    if (opts._gitignore) {
-      files = toRelative(opts.cwd, files)
-      if (os.platform() === 'win32') files = toUnix(files)
-      files = opts._gitignore.filter(files)
-      files = toAbsolute(opts.cwd, files)
-      if (os.platform() === 'win32') files = toWin32(files)
-    }
-
-    // undocumented – do not use (used by bin/cmd.js)
-    if (opts._onFiles) opts._onFiles(files)
+     // undocumented – do not use (used by bin/cmd.js)
+    if (opts._onFiles) opts._onFiles(allFiles)
 
     var result
     try {
-      result = new eslint.CLIEngine(opts._config).executeOnFiles(files)
+      result = new eslint.CLIEngine(opts._config).executeOnFiles(allFiles)
     } catch (err) {
       return cb(err)
     }
     return cb(null, result)
   })
+
 }
 
 function parseOpts (opts) {
   if (!opts) opts = {}
-  opts = extend(opts)
+  opts = extend(opts, DEGLOB_OPTIONS)
   opts._config = extend(DEFAULT_CONFIG)
+  opts.configKey = 'standard'
 
   if (!opts.cwd) opts.cwd = process.cwd()
 
-  opts._ignore = DEFAULT_IGNORE_PATTERNS.slice(0) // passed into glob
-  opts._gitignore = ignorePkg()
-
-  function addIgnorePattern (patterns) {
-    opts._ignore = opts._ignore.concat(patterns)
-    opts._gitignore.addPattern(patterns)
-  }
+  var ignore = DEFAULT_IGNORE_PATTERNS.slice(0) // passed into glob
 
-  if (opts.ignore) addIgnorePattern(opts.ignore)
-  if (opts.parser) useCustomParser(opts.parser)
```

Comment:
I think you accidentally removed this line. We still want to support custom parsers!